File: core/src/main/scala/nclogic/model/expr/Change.scala
package nclogic.model.expr

case class Change(e: Expr, level: Int) extends Expr {
  override lazy val toString = {
    val d = if (level > 1) s"^$level" else ""
    s"C$d($e)"
  }

  override lazy val boolString: String = ???



  lazy val simplify: Expr = e match {
    case Not(x) => Change(x, level).simplify
    case _ =>
      val s = if (level == 1) e.simplify else Change(e, level - 1).simplify
      ((s & !N(s)) | (!s & N(s))).simplify
  }
}

File: javaapi/src/main/java/nclogic/java/model/expr/Change.java
package nclogic.java.model.expr;

public class Change extends Expr {

    private final nclogic.model.expr.Change change;
    private final Expr e;

    public Change(Expr e) {
        this.e = e;
        change = new nclogic.model.expr.Change(e.getScalaExpr());
    }

    public Change(nclogic.model.expr.Change change) {
        this.change = change;
        e = ToJavaExprConverter.convert(change.e());
    }

    public Expr getE() {
        return e;
    }

    @Override
    public nclogic.model.expr.Expr getScalaExpr() {
        return change;
    }
}
